test/core/crypto/hasher/hasher_test.cpp

```diff
@@ -0,0 +1,150 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "common/blob.hpp"
+#include "crypto/hasher/hasher_impl.hpp"
+
+using kagome::common::Buffer;
+
+/**
+ * Hasher fixture
+ */
+class HasherFixture : public testing::Test {
+ public:
+  ~HasherFixture() override = default;
+
+  HasherFixture() {
+    hasher = std::make_shared<kagome::hash::HasherImpl>();
+  }
+
+  /**
+   * useful function for convenience
+   */
+  template <int size>
+  static Buffer blob2buffer(const kagome::common::Blob<size> &blob) noexcept {
+    Buffer out;
+    out.put({blob.data(), static_cast<long>(blob.size())});
+    return out;
+  }
+
+  static Buffer string2buffer(const std::string_view &view) noexcept {
+    Buffer out;
+    out.put(view);
+    return out;
+  }
+
+ protected:
+  std::shared_ptr<kagome::hash::Hasher> hasher;
+};
+
+/**
+ * @given empty source value
+ * @when Hasher::hashTwox_128 method is applied
+ * @then expected result obtained
+ */
+TEST_F(HasherFixture, twox_128_empty) {
+  auto hash = hasher->twox_128({});
+  std::vector<uint8_t> match = {153, 233, 216, 81, 55, 219, 70,  239,
+                                75,  190, 163, 54, 19, 186, 175, 213};
+
+  ASSERT_EQ(blob2buffer<16>(hash).toVector(), match);
+}
+
+/**
+ * @given some common source value
+ * @when Hasher::hashTwox_128 method is applied
+ * @then expected result obtained
+ */
+TEST_F(HasherFixture, twox_128_some) {
+  auto hash = hasher->twox_128({0x41, 0x42, 0x43, 0x44, 0x45, 0x46});
+  std::vector<uint8_t> match = {184, 65, 176, 250, 243, 129, 181, 3,
+                                77,  82, 63,  150, 129, 221, 191, 251};
+  ASSERT_EQ(blob2buffer<16>(hash).toVector(), match);
+}
+
+/**
+ * @given empty source value
+ * @when Hasher::hashTwox_256 method is applied
+ * @then expected result obtained
+ */
+TEST_F(HasherFixture, twox_256_empty) {
+  auto hash = hasher->twox_256({});
+  std::vector<uint8_t> match = {153, 233, 216, 81,  55,  219, 70,  239,
+                                75,  190, 163, 54,  19,  186, 175, 213,
+                                111, 150, 60,  100, 177, 243, 104, 90,
+                                78,  180, 171, 214, 127, 246, 32,  58};
+  ASSERT_EQ(blob2buffer<32>(hash).toVector(), match);
+}
+
+/**
+ * @given some common source value
+ * @when Hasher::hashTwox_256 method is applied
+ * @then expected result obtained
+ */
+TEST_F(HasherFixture, twox_256_some) {
+  // some value
+  auto hash = hasher->twox_256({0x41, 0x42, 0x43, 0x44, 0x45, 0x46});
+  std::vector<uint8_t> match = {184, 65,  176, 250, 243, 129, 181, 3,
+                                77,  82,  63,  150, 129, 221, 191, 251,
+                                33,  226, 149, 136, 6,   232, 81,  118,
+                                200, 28,  69,  219, 120, 179, 208, 237};
+  ASSERT_EQ(blob2buffer<32>(hash).toVector(), match);
+}
+
+/**
+ * @given empty source value
+ * @when Hasher::hashSha2_256 method is applied
+ * @then expected result obtained
```

Comment:
what is expected value? this comment is the same as the comment for the next test. But tests themselves are different. Please rewrite duplicate comments for these and other tests

Reply:
Or remove duplicate tests if they are checking the same thing

Reply:
One tests empty source value, second - some non-empty value. Or you consider one test is enough?

Reply:
One test per hash method left, others removed.